src/app/api/search.api.ts

```diff
@@ -45,21 +51,21 @@ export class SearchApi {
         private getDatasetLineageGQL: GetDatasetLineageGQL,
     ) {}
 
-    public pageInfoInit(): PageInfoInterface {
+    public pageInfoInit(): PageBasedInfo {
         return {
             hasNextPage: false,
             hasPreviousPage: false,
             totalPages: 0,
+            currentPage: 0,
         };
     }
 
-    // Search query that returs high-level dataset information for displaying the dataaset badge
+    // Search query that returns high-level dataset information for displaying the dataset badge
     public searchOverview(
         searchQuery: string,
         page = 0,
         perPage = 10,
-    ): Observable<SearchDatasetsOverviewQuery> {
-        // @ts-ignore
+    ): Observable<{search: Search} | undefined> {
```

Comment:
Continue using GraphQL query type like `SearchDatasetsOverviewQuery` here and in other observables within this file.